src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,24 +8,44 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, some street,23 ,234";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should have block, postal code, street and unit";
+    public static final String LIST_OF_4_CONSTRAINT_REGEX = "([a-zA-Z0-9#  -]+)(,[a-zA-Z0-9#  -]+){3}";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final int BLOCK_INDEX = 1;
+    public static final int POSTALCODE_INDEX = 2;
+    public static final int STREET_INDEX = 0;
+    public static final int UNIT_INDEX = 3;
 
     public final String value;
```

Comment:
Unacceptable to retain the variable value here for convenience. Unless there is a requirement that the formatting is different in the user-specified string, no need to retain. Else retaining original will result in duplication.